tools/prepare_data.R

```diff
@@ -8,25 +10,52 @@ forecast_date = Sys.Date()
 
 portalr::download_observations(release_only = FALSE)
 moons = get_moon_data()
+current_moons <- select(moons, newmoonnumber, newmoondate, period, censusdate)
+current_moons$newmoondate <- as.Date(as.character(current_moons$newmoondate))
 #get dates of 12 new moons following newmoon of interest
-future_moons = get_future_moons(moons)
+future_moons <- get_future_moons(moons)
+total_moons <- rbind(current_moons, future_moons)
 
-#Beginning and end of the forecast timeperiod
-most_recent_newmoon = moons$newmoonnumber[which.max(moons$period)]
-first_forecast_newmoon=most_recent_newmoon+1
-last_forecast_newmoon=first_forecast_newmoon + 11
-forecast_newmoons = first_forecast_newmoon:last_forecast_newmoon
-forecast_months = future_moons$month[future_moons$newmoonnumber %in% forecast_newmoons]
-forecast_years = future_moons$year[future_moons$newmoonnumber %in% forecast_newmoons]
 
 rodent_data = get_rodent_data(moons, forecast_date)
-weather_data = get_weather_data(moons, rodent_data$all, lag=6)
+weather_data = weather("newmoon",fill=TRUE) %>% dplyr::ungroup() %>%
+                dplyr::select("mintemp","maxtemp","meantemp","precipitation","newmoonnumber")
+incompletes <- which(is.na(weather_data$newmoonnumber))
+if(length(incompletes) > 0){
+  weather_data <- weather_data[-incompletes, ]
+}
+ndvi_data <- ndvi("newmoon", fill = TRUE)
+
+# Beginning and end of the forecast timeperiod
+most_recent_newmoon <- moons$newmoonnumber[max(which(total_moons$newmoondate < forecast_date))]
+most_recent_rodent_period <- max(tail(rodent_data$all, 1)$period, tail(rodent_data$control, 1)$period)
+most_recent_rodent_newmoon <- moons$newmoonnumber[which(moons$period == most_recent_rodent_period)]
+most_recent_covariate_newmoon <- min(tail(weather_data, 1)$newmoonnumber, tail(ndvi_data, 1)$newmoonnumber)
+last_forecast_newmoon <- most_recent_newmoon + 12
+first_forecast_covariate_newmoon <- most_recent_covariate_newmoon + 1
+first_forecast_rodent_newmoon <- most_recent_rodent_newmoon + 1
+covariate_forecast_newmoons <- first_forecast_covariate_newmoon:last_forecast_newmoon
+rodent_forecast_newmoons <- first_forecast_rodent_newmoon:last_forecast_newmoon
+covariate_nm_dates <- total_moons$newmoondate[total_moons$newmoonnumber %in% covariate_forecast_newmoons]
+rodent_nm_dates <- total_moons$newmoondate[total_moons$newmoonnumber %in% rodent_forecast_newmoons]
+covariate_forecast_months <- as.numeric(format(covariate_nm_dates, "%m"))
+covariate_forecast_years <- as.numeric(format(covariate_nm_dates, "%Y"))
+rodent_forecast_months <- as.numeric(format(rodent_nm_dates, "%m"))
+rodent_forecast_years <- as.numeric(format(rodent_nm_dates, "%Y"))
```

Comment:
I know what the problem is here, and the need for rodent dates and covariate dates is still confusing. I think we might need more explanation.